### PR TITLE
[Azure Search] Regenerate SDK for access condition methods

### PR DESCRIPTION
**Pull request by arv100kri**
1. Regenerate from https://github.com/Azure/azure-rest-api-specs/pull/7777 to pull in access condition methods for skillsets.

2. Add `MySQL` data source type which is in preview.

### CODE REVIEW

**Comment by arv100kri** [comment]
(I hit create pull request instead of create draft pull request by mistake.
Won't check this in until I check in the swagger PR and regenerate from master)

File: sdk/search/Microsoft.Azure.Search.Service/src/Customizations/DataSources/Models/DataSource.Customization.cs [hunk 1]
```diff
@@ -251,6 +251,48 @@ public partial class DataSource : IResourceWithETag
             };
         }
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search.Service/src/Customizations/DataSources/Models/DataSource.Customization.cs]: Looks like you're missing access condition tests for skillsets.

**Review by bitsforthought — CHANGES_REQUESTED** [review]
Missing test coverage, and maybe custom code too? Please ensure the ETag-related surface area for skillsets is at par with the other resources.

**Comment by arv100kri** [comment]
> Missing test coverage, and maybe custom code too? Please ensure the ETag-related surface area for skillsets is at par with the other resources.

Yup, just added some more tests and ensured that the surface area is consistent (compared with DataSource for example)

File: sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Skillsets/ISkillsetsOperations.Customization.cs [hunk 2]
```diff
@@ -13,31 +13,47 @@ namespace Microsoft.Azure.Search
...
+        /// A response object containing the response body and response headers.
+        /// </returns>
         Task<AzureOperationResponse<Skillset>> CreateOrUpdateWithHttpMessagesAsync(Skillset skillset, SearchRequestOptions searchRequestOptions = default(SearchRequestOptions), AccessCondition accessCondition = default(AccessCondition), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
 
         /// <summary>
         /// Determines whether or not the given skillset exists in the search service.
         /// </summary>
         /// <param name="skillsetName">
-        /// The name of the skillset.
+        /// The name of the data source.
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Skillsets/ISkillsetsOperations.Customization.cs]: skillset

**Review by bitsforthought — APPROVED** [review]
Minor doc bug, otherwise looks good.

**Review by bitsforthought — APPROVED** [review]

**Review by bleroy — APPROVED** [review]

File: sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Skillsets/SkillsetsOperations.Customization.cs [hunk 3]
```diff
@@ -12,13 +12,76 @@ namespace Microsoft.Azure.Search
...
+        /// <see href="https://docs.microsoft.com/rest/api/searchservice/Update-Skillset" />
+        /// </summary>
+        /// <param name='skillset'>
+        /// The definition of the skillset to create or update.
+        /// </param>
+        /// <param name='searchRequestOptions'>
+        /// Additional parameters for the operation.
+        /// </param>
+        /// <param name='accessCondition'>
+        /// Additional parameters for the operation.
```
[the remark this answers is not in the document]
Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Skillsets/SkillsetsOperations.Customization.cs]: Right, this seems to be the case for other customizations as well, so I thought I'll just use it as-is... Doesn't seem too egregious, so I'll keep it

File: sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Skillsets/SkillsetsOperations.Customization.cs [hunk 4]
```diff
@@ -12,13 +12,76 @@ namespace Microsoft.Azure.Search
...
+        /// The cancellation token.
+        /// </param>
+        /// <exception cref="CloudException">
+        /// Thrown when the operation returned an invalid status code.
+        /// </exception>
+        /// <exception cref="Microsoft.Rest.SerializationException">
+        /// Thrown when unable to deserialize the response.
+        /// </exception>
+        /// <exception cref="Microsoft.Rest.ValidationException">
+        /// Thrown when a required parameter is null.
```
[the remark this answers is not in the document]
Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Skillsets/SkillsetsOperations.Customization.cs]: Same - followed the pattern from elsewhere

**Pull request merged**